### PR TITLE
Finalize action

### PR DESCRIPTION
**Pull request by YevheniiSemendiak**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+FROM python:3.8.10-slim-buster
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-        libyaml-cpp-dev=0.6.2-4 libyaml-dev=0.2.1-1 && \
+    libyaml-cpp-dev=0.6.2-4 libyaml-dev=0.2.1-1 default-libmysqlclient-dev=1.0.5 gcc=4:8.3.0-1 && \
     apt-get clean && rm -rf /var/lib/apt/lists/*
 
-RUN pip install psycopg2-binary==2.8.6 mlflow[extras]==$MLFLOW_VERSION
+COPY requirements.txt requirements.txt
```
Comment (anayden) [inline Dockerfile:8]: I would do `COPY requirements.txt /requirements.txt` to make destination path more explicit

File: README.md [hunk 2]
```diff
@@ -1 +1,155 @@
-# mlflow
\ No newline at end of file
+# Run your personal MLFlow server on top of Neu.ro platform with neuro-flow
+
+This is a [`neuro-flow`](https://github.com/neuro-inc/neuro-flow) action launching an instance of [MLFlow tracking server](https://www.mlflow.org/docs/latest/tracking.html).
+You might use it to track your ML experiments, model trainings as well as deploying those models to production using our [MLFlow2Seldon intrgration](https://github.com/neuro-inc/mlops-k8s-mlflow2seldon)
```
Comment (anayden) [inline README.md:4]: `track your ML experiments, model trainings ` → `track your ML experiments AND model trainings`
`those models` → `the models`
`intrgration` → `integration

File: README.md [hunk 3]
```diff
@@ -1 +1,155 @@
...
+
+#### Examples
+
+Postgres server as a job within the same project:
+```
+args:
+	backend_store_uri: postgresql://postgres:password@${{ inspect_job('postgres').internal_hostname_named }}:5432
+```
+
+SQLite persistet on a platform disk.
```
Comment (anayden) [inline README.md:25]: `persistet` → `persistent`

File: README.md [hunk 4]
```diff
@@ -1 +1,155 @@
...
+This also implies adding the respective disk with the mount part `/some/path` to the `volumes` argument.
```
Comment (anayden) [inline README.md:26]: `mount part` → `mount path`

File: README.md [hunk 5]
```diff
@@ -1 +1,155 @@
...
+```
+args:
+    backend_store_uri: sqlite:///some/path/mlflow.db
+```
+
+Regular file. In this case "MLFlow registered models" will not work.
```
Comment (anayden) [inline README.md:32]: `will not work`? I think this requires some explanation. It's unclear to me :)

File: README.md [hunk 6]
```diff
@@ -1 +1,155 @@
...
+        }
+      ) }}"
+```
+
+### `http_auth`
+
+Whether to use HTTP authentication for Jupyter or not. `"False"` by default.
+
+#### Example
+Enable http_auth via setting it to True. In this case your training job should be able to communicate with hidden behind SSO MLFlow.
```
Comment (anayden) [inline README.md:93]: `communicate with hidden behind SSO MLFlow` — what do you mean by that?
I would at least rephrase as `communicate with MLFlow guarded by an SSO solution`, but this is not clear to me either :)

File: README.md [hunk 7]
```diff
@@ -1 +1,155 @@
...
+Enable http_auth via setting it to True. In this case your training job should be able to communicate with hidden behind SSO MLFlow. 
+```
+args:
+    http_auth: "True"
+```
+
+
+### `life_span`
+
+How log should the MLFlow server job run. `"10d"` by default.
```
Comment (anayden) [inline README.md:102]: `log` → `long`

File: README.md [hunk 8]
```diff
@@ -1 +1,155 @@
...
+
+```
+args:
+    preset: cpu-small
+```
+
+### `extra_params`
+
+Additional parameters transferred to the `mlflow server` command. `""` by default.
+Checkout the list of parameters via `mlflow server --help`.
```
Comment (anayden) [inline README.md:148]: `checkout` → `check`

File: action.yaml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    default: "[]"
+    descr: Additional volumes for the server job
+  envs:
+    default: "{}"
+    descr: Additional environment variables for the server job
+  http_auth:
+    default: "False"
+    descr: |
+      Whether to enable Neu.ro platform-based authentication or not.
+      If disabled (by default) - your WebDAV will not be protected by Neu.ro SSO.
```
Comment (anayden) [inline action.yaml:34]: `WebDAV` → `MLFlow server`

File: action.yaml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+      It has no impact on `rclone serve webdav` parameters though.
+  life_span:
+    default: "10d"
+    descr: |
+      How log should the MLFlow server job run
```
Comment (anayden) [inline action.yaml:39]: `log` → `long`

File: action.yaml [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+  port:
+    default: "5000"
+    descr: |
+      MLFlow server port.
+      Useful if you want to access the server within the cluster, for instance, from another job.
```
Comment (anayden) [inline action.yaml:44]: I don't understand the second line :( What do you mean by "useful if…"?

File: action.yaml [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+  job_name:
+    default: ""
+    descr: |
+      The name of MLFlow server.
```
Comment (anayden) [inline action.yaml:48]: `The name of MLFlow server JOB`

**Review by anayden — CHANGES_REQUESTED** [review]
Thanks for making this a neuro-flow action!
There's a bunch of wording comments and unclear phrases, but nothing critical.

File: action.yaml (hunk 12 above)
Comment (YevheniiSemendiak) [reply action.yaml]: Done in 54696036bdd

File: action.yaml (hunk 11 above)
Comment (YevheniiSemendiak) [reply action.yaml]: Removed that confusing line, rephrased this descr to `Port, which the MLFlow server listens will listen to and the platform exposes.` in 54696036bdd

File: action.yaml (hunk 10 above)
Comment (YevheniiSemendiak) [reply action.yaml]: Yeah, thank you a lot! Fixed in 54696036bdd

File: action.yaml (hunk 9 above)
Comment (YevheniiSemendiak) [reply action.yaml]: True, now you see from where I took the skeleton of this action :D 
Fixed in 54696036bdd

File: README.md (hunk 8 above)
Comment (YevheniiSemendiak) [reply README.md]: Thank you, fixed in 54696036bdd

File: README.md (hunk 7 above)
Comment (YevheniiSemendiak) [reply README.md]: Thanks! Fx in 54696036bdd

File: README.md (hunk 6 above)
Comment (YevheniiSemendiak) [reply README.md]: Rephrased to `In this case your training job should be able to communicate with MLFlow guarded by the Neu.ro platform authentication solution.` in  54696036bdd

File: README.md (hunk 4 above)
Comment (YevheniiSemendiak) [reply README.md]: fixed in  5469603

File: README.md (hunk 3 above)
Comment (YevheniiSemendiak) [reply README.md]: oki, done in  5469603

File: README.md (hunk 2 above)
Comment (YevheniiSemendiak) [reply README.md]: sure, fixed in   5469603

File: Dockerfile (hunk 1 above)
Comment (YevheniiSemendiak) [reply Dockerfile]: good idea, moved this file in /tmp/requiements.txt and removed it after use in  5469603

**Review by anayden — APPROVED** [review]
Thank you for your quick fixes!

**Pull request merged**